rancher/build.go

```diff
@@ -17,13 +17,14 @@ type Uploader interface {
 	Name() string
 }
 
-func getUploader() Uploader {
+func getUploader(c *Context) Uploader {
 	// This should do some factory magic
-	return &S3Uploader{}
+	return c.Uploader
 }
 
-func Upload(p *project.Project, name string) (string, string, error) {
-	uploader := getUploader()
+func Upload(c *Context, name string) (string, string, error) {
+	uploader := getUploader(c)
```

Comment:
check if uploader is nil and return err saying build is not supported.  Then you don't even need a change in executor, just set uploader to nil